From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Wed, 29 Jun 2016 17:39:43 +0300
Subject: [PATCH] be2net: signedness bug in be_msix_enable()
Patch-mainline: v4.8-rc1
Git-commit: 6fde0e63eccbaf21fa278b240b8129fec14b864b
References: bsc#963844 FATE#320192

"num_vec" needs to be signed for the error handling to work.

Fixes: e261768e9e39 ('be2net: support asymmetric rx/tx queue counts')
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Acked-by: Sathya Perla <sathya.perla@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/emulex/benet/be_main.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -3251,8 +3251,9 @@ static void be_msix_disable(struct be_ad
 
 static int be_msix_enable(struct be_adapter *adapter)
 {
-	unsigned int i, num_vec, max_roce_eqs;
+	unsigned int i, max_roce_eqs;
 	struct device *dev = &adapter->pdev->dev;
+	int num_vec;
 
 	/* If RoCE is supported, program the max number of vectors that
 	 * could be used for NIC and RoCE, else, just program the number
